tests/testsuite/cargo_alias_config.rs

```diff
@@ -153,13 +153,14 @@ fn alias_override_builtin_alias() {
         .build();
 
     p.cargo("b")
-        .with_stderr(
+        .with_stderr(&format!(
             "\
 [COMPILING] foo v0.5.0 ([..])
 [FINISHED] dev [unoptimized + debuginfo] target(s) in [..]
-[RUNNING] `target/debug/foo[EXE]`
+[RUNNING] `target/{}/debug/foo[EXE]`
```

Comment:
Does it still get copied to `target/debug/foo`? That is where pretty much everyone expects it to be in case of not cross-compiling. I think the examples and tests dir should also be copied. The deps and build dirs are more implementation details I think, so they don't have to be copied.

Reply:
> Does it still get copied to `target/debug/foo`?

No

> That is where pretty much everyone expects it to be in case of not cross-compiling. I think the examples and tests dir should also be copied.

Based on previous discussions the goal is to ensure that non-cross compilation cases are effectively identical to cross compilation cases, so not sure this is desirable.

> The deps and build dirs are more implementation details I think, so they don't have to be copied.

Isn't that also the case here? This isn't really an installation directory from my understanding so a specific path here generally shouldn't be important AFAIU.

Reply:
When invoking cargo through a different build system or even when automatically building, you have to rely on this. `cargo install` can't be used for staticlibs and dylibs. It can be used for executables, but I don't think many people do in the context of building programs to be used outside of the current system.

Reply:
> When invoking cargo through a different build system or even when automatically building, you have to rely on this.

Shouldn't one set an installation directory there instead? That way it should still work either way. Here's [an example](https://patchwork.ozlabs.org/project/buildroot/patch/20200220160119.3407-1-patrick.havelange@essensium.com/) in buildroot of how this sort of change(still a WIP) is being made.

Reply:
That doesn't work for libraries (staticlib/cdylib). Those can't be installed.

Reply:
> Those can't be installed.

Hmm, seems that's a known cargo bug, we probably should fix that. Normally meta cross build systems(like [buildroot](https://buildroot.org/)) use standard install features from the packages specific build systems(such as autotools or meson) to install libraries to a staging directory for this sort of thing, cargo really should provide an equivalent feature there.

Reply:
Generally "staticlib/cdylib" libraries like this are also expected to provide pkg-config `.pc` files when installed as well so that other software knows how to link against them properly, so cargo probably needs a way to generate those automatically as well when installing.